epidose/device/wps_scanner_d.sh

```diff
@@ -112,17 +110,110 @@ wps_connect()
   return "$exit_code"
 }
 
+# Check whether the device has obtained
+# an IP address from network's dhclient
+# preconditions: WiFi should be turned on
+# Returns 0 if the device has an IP address
+# or 1 if not
+check_if_ip_obtained()
+{
+  get_ip_address=$(ifconfig wlan0 | grep "inet ")
+
+  # Check if device has an IP address
+  if [ -z "$get_ip_address" ] ; then
+    log "Failed to connected to network"
+    return 1
+  fi
+
+  # If the device has received an IP address,
+  # then return 0
+  log "Successfully connected to network"
+  return 0
+}
+
+# Try to establish network connection
+# with the configurations stored in
+# the wpa_supplicant.conf file
+# precondition: WiFi should be turned on
+# Returns 0 if the device has successfully
+# connected to a network or 1 if not
+connect_to_saved_network()
+{
+  # Check if network $1 is in range
+  get_network_if_in_range=$(wpa_cli -i wlan0 scan_results | grep "$1")
+  if [ -z "$get_network_if_in_range"  ]; then
+    log "Network $1 not in range"
+    return 1
+  fi
+
+  # Get $1 network's id, list_networks checks the /etc/wpa_supplicant/wpa_supplicant.conf file
+  get_network_id=$(wpa_cli list_networks | grep "$1" | cut -f 1)
```

Comment:
* Replace grep/cut with `awk '$2 ~ /^'$1'$/ {print $1}'` to ensure better matching.
* Remove `get_` from variable names: variables contain values they don't perform actions